airflow-ctl/src/airflowctl/api/operations.py

```diff
@@ -148,6 +150,33 @@ def __init_subclass__(cls, **kwargs):
             if callable(value):
                 setattr(cls, attr, _check_flag_and_exit_if_server_response_error(value))
 
+    def return_all_entries(
+        self,
+        *,
+        path: str,
+        total_entries: int,
+        data_model: type[BaseModel],
+        entry_list: list,
+        offset: int = 0,
+        limit: int = 50,
+        params: dict | None = None,
+        **kwargs,
+    ) -> list | ServerResponseError:
+        if params is None:
+            params = {}
+        params.update({"limit": limit}, **kwargs)
```

Comment:
Probably not consequential, but updating `params` in-place is somewhat surprising. I’d do something like

```python
shared_params = {"limit": limit, **(params or {}), **kwargs}

while ...:
    loop_params = {**shared_params, "offset": offset}
```

Also, the kwargs feature seems somewhat unnecessary. I would just get rid of it and require everything passed inside params.